x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/UnresolvedRelation.java

```diff
@@ -71,7 +71,7 @@ protected NodeInfo<UnresolvedRelation> info() {
         return NodeInfo.create(this, UnresolvedRelation::new, table, frozen, metadataFields, indexMode, unresolvedMsg, commandName);
     }
 
-    public TableIdentifier table() {
+    public IndexPattern table() {
```

Comment:
I wonder if the method should be renamed as well? Or may be it is used in too many places

Reply:
Yeah, that name is just misleading at the moment. But I'd like to do this in another PR to keep the blast radius contained.

Reply:
Actually, I changed my mind. Let's go and have a little more readability and consistency.